src/plugins/dashboard/public/dashboard_container/external_api/dashboard_renderer.tsx

```diff
@@ -91,33 +92,41 @@ export const DashboardRenderer = forwardRef<AwaitingDashboardAPI, DashboardRende
         const dashboardFactory = embeddable.getEmbeddableFactory(
           DASHBOARD_CONTAINER_TYPE
         ) as DashboardContainerFactory & { create: DashboardContainerFactoryDefinition['create'] };
-        const container = (await dashboardFactory?.create(
+        const container = await dashboardFactory?.create(
           { id } as unknown as DashboardContainerInput, // Input from creationOptions is used instead.
           undefined,
           creationOptions,
           savedObjectId
-        )) as DashboardContainer;
+        );
 
         if (canceled) {
           container.destroy();
           return;
         }
 
+        if (isErrorEmbeddable(container)) {
```

Comment:
Shouldn't `setLoading(false);` get called if return is an error embeddable? The container finished loading, even if it loaded in an error state

Reply:
Good call, I didn't catch that because the error was rendered if it was present no matter if the loading state was true or not. I've straightened that out!